plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -214,7 +214,7 @@ public void registerTable(ConnectorSession session, SchemaTableName tableName, S
     @Override
     public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        if (!restSessionCatalog.dropTable(convert(session), toIdentifier(schemaTableName))) {
+        if (!restSessionCatalog.purgeTable(convert(session), toIdentifier(schemaTableName))) {
```

Comment:
@ebyhr Rather than change the current behavior, can we make this a configuration property?  There are many situations where you don't want drop to destroy data and I'd rather be explicit about the intent rather than have systems ignore the request (like is done in many cases with Hive Metastore).

Reply:
We've been moving towards solving that situation with an `unregister_table` procedure: https://github.com/trinodb/trino/issues/15853

`DROP TABLE` is an unrecoverable operation across all the other Trino connectors, so I think that a separate procedure like this will be clearer to users than a config property that gets set at cluster creation.

Reply:
Hopefully the symmetrical naming with the `register_table` procedure will make it more intuitive too